message_app_client/src/components/conversation.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+
+class ConversationCard extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            lastMessageContent: null,
+            conversationStatus: null,
```

Comment:
What is the value of `conversationStatus`? Is it `away` or `online` like in the old version?

About that old version, I forgot to address that, we do not display the status there. It is actually the other participant's status, right?

The status of a conversation that we can consider here, is probably whether or not the last message is from the participant, and whether that last message has been read by the current user.
So, for this use case, something like `hasUnseenMessage` with True or False values might be a decent choice.

Reply:
Yes, the value for `conversationStatus` is either `active` or `away`, and the status is the other participant's status. I put the status next to the conversation title because we don't have a way to display the user's status yet. We may display it as a green/black point next to the conversation title later on (like other message apps). However, for now, we can just display it as text.

I will implement the `seen` function after we finished migrating to React. 👍

Reply:
So the name should be `otherParticipantStatus` instead of `conversationStatus`, and we will only have this property if the number of people in the conversation is 2, right?

Reply:
Make sense. I will change the name of the property to 'otherParticipantStatus'

For the conversation which has more than 2 users, there are two choices:
- As you said, we can delete the property if the number of users is more than 2
- Or, we can set the status as 'active', as long as there is at least one user in the group is 'active'(facebook's way).
- We can discuss this in our next meeting

Reply:
I would prefer to set the property to an empty string maybe, to indicate that it is not for use.
So, I choose the first option.